src/core/Akka.Persistence/Eventsourced.cs

```diff
@@ -455,6 +455,30 @@ public void DeleteMessages(long toSequenceNr)
             Journal.Tell(new DeleteMessagesTo(PersistenceId, toSequenceNr, Self));
         }
 
+        /// <summary>
+        /// An <see cref="Eventsourced"/> actor can request cleanup by deleting either a range of, or all persistent events.
+        /// For example, on successful snapshot completion, delete messages within a configurable <paramref name="snapshotAfter"/>
+        /// range that are less than or equal to the given <see cref="SnapshotMetadata.SequenceNr"/>
+        /// (provided the <see cref="SnapshotMetadata.SequenceNr"/> is &lt;= to <see cref="Eventsourced.LastSequenceNr"/>).
+        ///
+        /// Or delete all by using `long.MaxValue` as the `toSequenceNr`
+        /// {{{ m.copy(sequenceNr = long.MaxValue) }}}
+        /// </summary>
+        /// <param name="e"></param>
+        /// <param name="keepNrOfBatches"></param>
+        /// <param name="snapshotAfter"></param>
+        internal void InternalDeleteMessagesBeforeSnapshot(SaveSnapshotSuccess e, int keepNrOfBatches, int snapshotAfter)
```

Comment:
Like this - not user-friendly enough to make `public` but good enough for our internal uses of `Eventsourced`.